### ISSUE TITLE
How can I specify "database" in create_engine?

### ISSUE THREAD

**Issue by lyh710** [Teradata/sqlalchemy-teradata#30]
I have pandas dataframe (say df), which I want to write back to Teradata via df.to_sql(). I only have write access to one specific database in the host, say db_A. 

I did the below:

from sqlalchemy import create_engine
engine_str = 'teradata://'+ user +':' + pasw + '@'+ host + '/' + db_A+ '/' + '?authentication=LDAP' 
td_engine = create_engine(engine_str)
conn = td_engine.connect()
df.to_sql(name = db_A, con = td_engine, if_exists = 'replace', index = False)

Doing so, I can successfully establish connection to Teradata. However, the connection is to another database (say db_B) instead of db_A. The parameter "db_A" I pass to create_engine is ignored.

Due to the fact that I do not have write access to db_B, the code ends with error.

How can I force sqlalchemy to connect to the database of my choice?

**Comment by tserafim** [comment]
Always post the error message, it makes easier for people to help you.

After checking [to_sql documentation](https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.to_sql.html), i see that name argument is used for table name, and not database name.

Let´s say you want to write to table `MyTable` on database `db_b`.

Try:

    df.to_sql(name='db_b.MyTable', con=td_engine, if_exists = 'replace', index = False)

**Comment by lyh710** [comment]
Apologize for the not clear enough description. Let me try again:

There are 2 database in my Teradata host, db_A and db_B. 
My aim is to write a pandas dataframe (df) back to db_A, which will eventually create a table as db_A.MyTable.

I create the engine by:
engine_str = 'teradata://'+ user +':' + pasw + '@'+ host + '/' + db_A+ '/' + '?authentication=LDAP'
 td_engine = create_engine(engine_str)
 conn = td_engine.connect()

This part runs fine, no error. However, when I try to select data from db_A (to test my connection is in fact effective), I had below error:

td_engine.execute('select * from TestTable)

DatabaseError: (teradata.api.DatabaseError) (3807, "[42S02] [Teradata][ODBC Teradata Driver][Teradata Database] Object TestTable does not exist. ") [SQL: 'select * from TestTable']

The reason of this error is because the engine is now connected to df_B, and of course TestTable does no exist. This error can be fixed by:
td_engine.execute('select * from df_A.TestTable)

I can not understand why the engine is connecting to df_B, because I have specified db_A when creating it. This is where I need help.

And naturally, when I call
df.to_sql(name = 'MyTable', con = td_engine, if_exists = 'replace', index = False)

I had the below error:
DatabaseError: (teradata.api.DatabaseError) (3524, '[42000] [Teradata][ODBC Teradata Driver][Teradata Database] The user does not have CREATE TABLE access to database df_B. ')

Again, this is clearly showing the engine is connecting to df_B, regardless I have specified df_A when creating it.

I then tried:
df.to_sql(name = 'df_A.MyTable', con = td_engine, if_exists = 'replace', index = False)

And got the same error:
DatabaseError: (teradata.api.DatabaseError) (3524, '[42000] [Teradata][ODBC Teradata Driver][Teradata Database] The user does not have CREATE TABLE access to database df_B. ')

Overall, my thinking is I need to fix this by make sure the engine is connecting to df_A (the database I want) when created. And my problem is despite the fact the code runs fine, the database name I pass is ignored, and engine is connected to somewhat "default" database.
engine_str = 'teradata://'+ user +':' + pasw + '@'+ host + '/' + db_A+ '/' + '?authentication=LDAP'

Look forward to your help.

**Comment by sandan** [comment]
Try setting the database name you want in the schema parameter in the to_sql call. The name should just be the name of the table, it shouldn't be fully qualified. Also, once connecting to the database with the engine, try running the query "sel user" and "sel database" and inspect the values to see if they are expected.

**Comment by sakkammadam** [comment]
I was able to connect to the specific db using the create engine. This was what I used -create_engine("teradata://"+user+":"+passw+"@"+host+":22/"+db)
I did not use **LDAP**
